exist-core/src/main/java/org/exist/xquery/functions/array/ArrayType.java

```diff
@@ -382,6 +382,30 @@ public static Sequence flatten(Sequence input) throws XPathException {
         return flatten ? flatten(input, new ValueSequence(input.getItemCount() * 2)) : input;
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        for (int i = 0; i < vector.length(); i++) {
+            final Sequence value = vector.nth(i);
+            if (value == item) {
+                return true;
+            }
+            return value.containsReference(item);
```

Comment:
I think this will not work if the reference is not the first entry. I guess it should be:
```
if (value == item || value.containsReference(item)) {
  return true;
}
```

Reply:
AFAIK the logic you have suggested is exactly the same as what is already there. What am I missing?

Reply:
Okay fixed.